cmd/wallet-sdk-gomobile/activitylogger/mem/mem.go

```diff
@@ -0,0 +1,56 @@
+/*
+Copyright Avast Software. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+// Package mem contains a simple in-memory api.ActivityLogger implementation.
+package mem
+
+import (
+	"sync"
+
+	"github.com/trustbloc/wallet-sdk/cmd/wallet-sdk-gomobile/api"
+)
+
+// ActivityLogger is a simple in-memory api.ActivityLogger implementation.
+// New activities are appended to the array in the order that they are received.
+type ActivityLogger struct {
+	activities []*api.Activity
+	lock       sync.RWMutex
+}
+
+// NewActivityLogger returns a new ActivityLogger.
+func NewActivityLogger() *ActivityLogger {
+	return &ActivityLogger{}
+}
+
+// Log logs a single activity.
+// The activity is appended to the end of an internal array of activities.
+func (m *ActivityLogger) Log(activity *api.Activity) error {
+	m.lock.Lock()
+	defer m.lock.Unlock()
+
+	m.activities = append(m.activities, activity)
+
+	return nil
+}
+
+// Length returns the number of activities that this activity logger has logged.
+func (m *ActivityLogger) Length() int {
```

Comment:
What is the use of this function, underlying activity logger might just care about pushing this data model to store,

Reply:
This isn't part of the interface - it's only in the in-memory implementation. It's intended for the reference demo app so we have some way to collect activities and show them in the UI.

Reply:
(there is no underlying activity logger here)

Reply:
I also make use of this type in the tests